chromium_src/third_party/blink/renderer/modules/storage/dom_window_storage.cc

```diff
@@ -223,8 +223,11 @@ StorageArea* BraveDOMWindowStorage::ephemeralLocalStorage() {
   if (!namespaces)
     return nullptr;
 
-  auto storage_area =
-      namespaces->local_storage()->GetCachedArea(window->GetSecurityOrigin());
+  auto* controller = StorageController::GetInstance();
```

Comment:
no new tests are added here because this already works on desktop and browser tests don't run for android yet